types.go

```diff
@@ -55,16 +52,7 @@ type ReadSet struct {
 	Iterators []IteratorDescriptor
 }
 
-type WriteSet = MemDB
-
-func NewWriteSet() WriteSet {
-	return *NewMemDBNonConcurrent()
-}
-
-type (
-	MultiWriteSet = []WriteSet
-	MultiReadSet  = []ReadSet
-)
+type MultiReadSet = map[int]*ReadSet
```

Comment:
Can reduce size since it was possible to skip index in the middle before right?

Reply:
exactly, can save a little bit